include/git2/odb.h

```diff
@@ -171,6 +173,40 @@ GIT_EXTERN(int) git_odb_exists(git_odb *db, const git_oid *id);
 GIT_EXTERN(int) git_odb_exists_prefix(
 	git_oid *out, git_odb *db, const git_oid *short_id, size_t len);
 
+typedef struct git_odb_expand_id {
+	/** The object ID to expand */
+	git_oid id;
+
+	/** The length of the object ID */
+	unsigned short length;
+
+	/** The type of the object (optionally) */
+	git_otype type;
```

Comment:
Ah yes, I see the code. But I don't think we have tests for the zero, just for `_ANY`.